flink-java/src/main/java/org/apache/flink/api/java/operators/PartitionOperator.java

```diff
@@ -148,9 +147,6 @@ else if (pKeys instanceof Keys.SelectorFunctionKeys) {
 			}
 			
 		} 
-		else if (pMethod == PartitionMethod.RANGE) {
```

Comment:
Yes, that's my intention. Range partitioning is often used to produce fully sorted output. This does not work if the partition keys have no defined sort order.